File: notification/java/src/test/java/com/nublic/notification/java/NotificationTest.java
// Copyright (c) 2011, Nublic
/**
 * @author David Navarro Estruch 
 */
package com.nublic.notification.java;

// import static org.junit.Assert.*;

import org.junit.After;
import org.junit.AfterClass;
import org.junit.Before;
import org.junit.BeforeClass;
import org.junit.Test;

/**
 * @author david
 *
 */
public class NotificationTest {

	/**
	 * @throws java.lang.Exception
	 */
	@BeforeClass
	public static void setUpBeforeClass() throws Exception {
	}

	/**
	 * @throws java.lang.Exception
	 */
	@AfterClass
	public static void tearDownAfterClass() throws Exception {
	}

	/**
	 * @throws java.lang.Exception
	 */
	@Before
	public void setUp() throws Exception {
	}

	/**
	 * @throws java.lang.Exception
	 */
	@After
	public void tearDown() throws Exception {
	}

	/**
	 * Test method for {@link com.nublic.notification.java.Notification#newMessage(java.lang.String, java.lang.String, java.lang.String, java.lang.String)}.
	 */
	@Test
	public void testNewMessage() {
		Notification n = new Notification();
		n.newMessage("appJava","","Some thing appened", "NOTICE");
		
	}

}

File: notification/main/src/nublic_notification/notification_test.py
#!/usr/bin/python

'''
Created on 15/08/2010

@author: David Navarro Estruch
@copyright: 2011 Nublic
'''

import unittest
from notification import Notification, new_message

from elixir import create_all, setup_all, drop_all


class testNotification(unittest.TestCase):
    """
    A test class for the Notification module.
    """

    def __init__(self, methodName='runTest'):
        # Clean and create database
        setup_all(create_tables=True)
        drop_all()
        create_all()
        unittest.TestCase.__init__(self, methodName=methodName)

    def setUp(self):
        """
        set up data used in the tests.
        setUp is called before each test function execution.
        """
        pass

    def testNewMessage(self):
        new_message("app", "user1", "level", "CriticalMessage")
        notice = Notification.get_by(app="app")
        self.assertEqual(notice.app, "app",
                         "Value retrieved should be the stored")
        self.assertEqual(notice.user, "user1",
                         "Value retrieved should be the stored")
        self.assertEqual(notice.level, "level",
                         "Value retrieved should be the stored")
        self.assertEqual(notice.text, "CriticalMessage",
                         "Value retrieved should be the stored")
        #self.assertEqual(self.blogger.get_title(), title)


#
#def suite():
#
#    suite = unittest.TestSuite()
#
#    suite.addTest(unittest.makeSuite(testBlogger))
#
#    return suite
#    pass
#
#if __name__ == '__main__':
#    unittest.main()
